.github/workflows/mvn-verify-check.yml

```diff
@@ -53,7 +53,15 @@ jobs:
         id: generateCacheKey
         run: |
           set -x
-          cacheKey="${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}-${{ github.event.pull_request.base.ref }}-$(date +'%Y-%m-%d')"
+          jniVer=$(mvn help:evaluate -q -pl dist -Dexpression=spark-rapids-jni.version -DforceStdout)
+          privateVer=$(mvn help:evaluate -q -pl dist -Dexpression=spark-rapids-private.version -DforceStdout)
+          jniTimestamp=$(curl -s -H "Accept: application/json" \
+           "https://oss.sonatype.org/service/local/artifact/maven/resolve?r=snapshots&g=com.nvidia&a=spark-rapids-jni&v=${jniVer}&c=&e=jar&wt=json" \
+            | jq .data.snapshotTimeStamp) || $(date +'%Y-%m-%d')
+          privateTimestamp=$(curl -s -H "Accept: application/json" \
+           "https://oss.sonatype.org/service/local/artifact/maven/resolve?r=snapshots&g=com.nvidia&a=rapids-4-spark-private_2.12&v=${privateVer}&c=&e=jar&wt=json" \
+            | jq .data.snapshotTimeStamp) || $(date +'%Y-%m-%d')
+          cacheKey="${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}-${{ github.event.pull_request.base.ref }}-${jniTimestamp}-${privateTimestamp}"
```

Comment:
There is quite a bit of repetition inside this stance and with the scala2.13 copy on top of it. It is also harder to read if a lot of shell code included in YAML. We might be able to avoid both of these issues by having a dedicated .sh file with a function that can be called for spark-rapids-jni, rapids-4-spark_private for 2.12 and 2.13 Scala